sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -0,0 +1,120 @@
+package com.microsoft.appcenter.utils.context;
+
+import android.support.annotation.VisibleForTesting;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+
+/**
+ * Utility to store and retrieve the latest authorization token.
+ */
+public class AuthTokenContext {
+
+    /**
+     * Unique instance.
+     */
+    private static AuthTokenContext sInstance;
+
+    /**
+     * Global listeners collection.
+     */
+    private final Collection<Listener> mListeners;
+
+    /**
+     * Current value of auth token.
+     */
+    private String mAuthToken;
+
+    /**
+     * Current value of account id.
+     */
+    private String mLastHomeAccountId;
+
+    /**
+     * Get unique instance.
+     *
+     * @return unique instance.
+     */
+    public static synchronized AuthTokenContext getInstance() {
+        if (sInstance == null) {
+            sInstance = new AuthTokenContext();
+        }
+        return sInstance;
+    }
+
+    /**
+     * A private constructor for the class.
+     */
+    private AuthTokenContext() {
+        mListeners = new LinkedHashSet<>();
+    }
+
+    /**
+     * Unsets singleton instance.
+     */
+    @VisibleForTesting
+    public static synchronized void unsetInstance() {
+        sInstance = null;
+    }
+
+    /**
+     * Adds listener to token context.
+     *
+     * @param listener listener to be notified of changes.
+     */
+    public synchronized void addListener(Listener listener) {
+        mListeners.add(listener);
+    }
+
+    /**
+     * Removes a specific listener.
+     *
+     * @param listener listener to be removed.
+     */
+    public synchronized void removeListener(Listener listener) {
+        mListeners.remove(listener);
```

Comment:
Not covered in tests

Reply:
@guperrot This will be covered once we merge Push PR into this branch. It uses `removeListener` and has tests to check if this listener is actually removed. But if not, I will add coverage after we merge all "child" PRs.